File: backend/schemas/notification.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, Optional
from uuid import UUID

class NotificationResponse(BaseModel):
    id: UUID
    booking_id: str
    notification_date: datetime
    notification_type: str
    message: str
    icon: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True

class NotificationListResponse(BaseModel):
    success: bool
    notifications: List[NotificationResponse]
    message: Optional[str] = None

File: src/models/notification.ts
export interface AppNotification {
    id: string;
    booking_id: string;
    notification_date: string;
    notification_type: string;
    message: string;
    icon?: string;
    isRead?: boolean; // We'll keep this for UI state even if not in DB yet
    created_at: string;
}
